tests/test_ProblemData.py

```diff
@@ -1314,6 +1314,49 @@ def test_max_lateness(solver: str):
     assert_equal(result.best.tasks[1].end, 2)
 
 
+def test_total_setup_time(solver: str):
+    """
+    Tests that the total setup time objective function is correctly optimized.
+    """
+    model = Model()
+
+    machine = model.add_machine()
+    tasks = [model.add_task() for _ in range(3)]
+
+    for idx in range(2):
+        first = tasks[idx]
+        second = tasks[idx + 1]
+        model.add_end_before_start(first, second)
+
+    for task in tasks:
+        model.add_mode(task, machine, duration=1)
+
+    # Setup times between tasks:
+    # - task0 -> task1 on machine: 1
+    model.add_setup_time(machine, tasks[0], tasks[1], duration=1)
+    # - task1 -> task2 on machine: 3
+    model.add_setup_time(machine, tasks[1], tasks[2], duration=3)
+
+    model.set_objective(weight_total_setup_time=2)
+
+    result = model.solve(solver=solver)
+
+    # The optimal solution has a makespan of 1 + 1 + 1 + 3 + 1 = 7,
+    # and the setup times are 1 and 3, respectively.
+    # The objective value is 2 * (1 + 3) = 8.
```

Comment:
Rewrite: This comment should explain why the tests below are expected to be true.

For instance, the makespan here is not relevant, because it's not part of the test, so it's not needed to comment about that.

What _is_ interesting to comment about is (1) how the solution looks like and (2) why it results in the assertion (in this case, objective of 8). I would write something along those lines:

"Tasks 0, 1 and 2 are scheduled consecutively because of the precedence constraints, so the setup times are 1 and 3, respectively. Combined with an objective weight of two, the objective value is 2 * (1 + 3) = 8."